services/filemanager/filemanager.go

```diff
@@ -102,7 +102,7 @@ func GetProviderConfigForBackupsFromEnv(ctx context.Context) map[string]interfac
 		providerConfig["iamRoleArn"] = config.GetString("BACKUP_IAM_ROLE_ARN", "")
 		if providerConfig["iamRoleArn"] != "" {
 			backendconfig.DefaultBackendConfig.WaitForConfig(ctx)
-			providerConfig["externalId"] = backendconfig.DefaultBackendConfig.GetWorkspaceIDForWriteKey("")
+			providerConfig["externalId"] = backendconfig.DefaultBackendConfig.Identity().ID()
```

Comment:
Note: As discussed, we can not use a namespace for `externalId`. This is of course outside the scope of this PR, but we need a reminder to support externId in multitenant environments.